packetbeat/pb/event.go

```diff
@@ -416,8 +416,17 @@ func marshalStruct(m mapstr.M, key string, val reflect.Value) error {
 	}
 
 	typ := val.Type()
+	// pre-emptively handle time
+	if reflect.TypeOf(time.Time{}) == typ {
```

Comment:
Was there some test failure that prompted this addition? If so, consider adding this case to the unit tests.